rotkehlchen/tasks/calendar.py

```diff
@@ -178,20 +179,109 @@ def maybe_create_ens_reminders(database: DBHandler) -> None:
         for secs_before in (WEEK_IN_SECONDS, DAY_IN_SECONDS)
     ])
 
-    with database.conn.write_ctx() as write_cursor:
-        database.set_static_cache(
-            write_cursor=write_cursor,
-            name=DBCacheStatic.LAST_CREATE_REMINDER_CHECK_TS,
-            value=current_ts,
+    if len(failed_to_add) == 0:
+        return
+
+    log.error(  # failed_to_add is a list of calendar_identifier that were passed as identifiers of reminder entry above  # noqa: E501
+        f"""Failed to add the ENS expiry reminders for {', '.join([
+            entry.name for entry in db_calendar.query_calendar_entry(CalendarFilterQuery.make(
+                identifiers=failed_to_add,
+            ))['entries']
+        ])}""",
+    )
+
+
+def maybe_create_locked_crv_reminders(database: DBHandler, current_ts: Timestamp) -> None:
+    """Check for lock CRV in vote escrow history events and create reminders if needed."""
+    db_history_events = DBHistoryEvents(database=database)
+    with database.conn.read_ctx() as cursor:
+        crv_events = db_history_events.get_history_events(
+            cursor=cursor,
+            has_premium=True,  # not limiting here
+            group_by_event_ids=False,
+            filter_query=EvmEventFilterQuery.make(
+                and_op=True,
+                counterparties=[CPT_CURVE],
+                event_types=[HistoryEventType.DEPOSIT],
+                event_subtypes=[HistoryEventSubType.DEPOSIT_ASSET],
+            ),
         )
 
+    if len(crv_events) == 0:
+        return
+
+    db_calendar = DBCalendar(database=database)
+    customizable_date = CustomizableDateMixin(database=database)
+    with database.conn.read_ctx() as cursor:
+        blockchain_accounts = database.get_blockchain_accounts(cursor=cursor)
+
+    crv_calender_entries: list[int] = []
+    for crv_event in crv_events:
+        if not (
+            crv_event.location_label is not None and
+            crv_event.extra_data is not None and
+            (locktime := crv_event.extra_data.get('locktime')) is not None and
+            (user_address := string_to_evm_address(crv_event.location_label)) in blockchain_accounts.get(  # noqa: E501
+                blockchain := ChainID.deserialize(crv_event.location.to_chain_id()).to_blockchain(),  # noqa: E501
+            ) and (locktime := Timestamp(locktime)) > current_ts  # those ending in the future
+        ):
+            continue
+
+        calendar_entry_description = f'Lock period for {crv_event.balance.amount} CRV in vote escrow ends on {customizable_date.timestamp_to_date(locktime)}'  # noqa: E501
```

Comment:
This entire pattern here seems like a copy pasta of the ens one. Can't this be abstracted into a common function. It's literally copy pasta. Also the comments.